presto-client/src/main/java/com/facebook/presto/client/FixJsonDataUtils.java

```diff
@@ -127,6 +127,12 @@ private static Object fixValue(TypeSignature signature, Object value)
         if (signature.isVarcharEnum()) {
             return String.class.cast(value);
         }
+        if (signature.isLongEnum()) {
+            if (value instanceof String) {
```

Comment:
Why would the value be string is the enum is long?

Reply:
that happens because of JSON serialization – in JSON, keys are always strings (this is why we have the same check a few lines below).